File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def solution(video_len, pos, op_start, op_end, commands):
    
    def minute_to_second(string):
        mm, ss = map(int, string.split(":"))
        return mm * 60 + ss
    
    def second_to_minute(time):
        mm = str(time // 60).zfill(2)
        ss = str(time % 60).zfill(2)
        return f"{mm}:{ss}"
    
    def opening_skip(pos):
        if minute_to_second(op_start) <= pos <= minute_to_second(op_end):
            return minute_to_second(op_end)
        else:
            return pos
        
    pos = minute_to_second(pos)
    
    for command in commands:
        pos = opening_skip(pos)
            
        if command == "prev":
            pos = max(0, pos-10)
        if command == "next":
            pos = min(minute_to_second(video_len), pos+10)
        
        pos = opening_skip(pos)
    
    return second_to_minute(pos)

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
import java.util.*;

class Solution {
    public int minuteToSecond(String s) { // 문자열->초
        String[] split_s = s.split(":");
        int mm = Integer.parseInt(split_s[0]);
        int ss = Integer.parseInt(split_s[1]);
        return mm * 60 + ss;
    }
    public String secondToMinute(int time) { // 초->문자열
        int mm = time / 60;
        int ss = time % 60;
        return String.format("%02d:%02d", mm, ss);
    }
    public int openingSkip(int op_start, int pos, int op_end) { // 오프닝 건너뛰기 확인
        if (op_start <= pos && pos <= op_end) {
            return op_end;
        }
        else {
            return pos;
        }
    }
    
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        int videoLen = minuteToSecond(video_len);
        int position =  minuteToSecond(pos);
        int opStart = minuteToSecond(op_start);
        int opEnd = minuteToSecond(op_end);
        
        for (String command: commands) {
            position = openingSkip(opStart, position, opEnd);
            
            if (command.equals("prev")) {
                position = Math.max(0, position - 10);
            }
            else if (command.equals("next")) {
                position = Math.min(videoLen, position + 10);
            }
            position = openingSkip(opStart, position, opEnd);
        }
        return secondToMinute(position);
    }
}